tests/migrations/test_writer.py

```diff
@@ -613,16 +613,21 @@ def test_migration_file_header_comments(self):
         })
         dt = datetime.datetime(2015, 7, 31, 4, 40, 0, 0, tzinfo=utc)
         with mock.patch('django.db.migrations.writer.now', lambda: dt):
-            writer = MigrationWriter(migration)
-            output = writer.as_string()
-
-        self.assertTrue(
-            output.startswith(
-                "# Generated by Django %(version)s on 2015-07-31 04:40\n" % {
-                    'version': get_version(),
-                }
-            )
-        )
+            for include_header in (True, False):
+                with self.subTest(include_header=include_header):
+                    writer = MigrationWriter(migration, include_header)
+                    output = writer.as_string()
+
+                    self.assertEqual(
+                        include_header,
+                        output.startswith(
+                            "# Generated by Django %s on 2015-07-31 04:40\n\n" % get_version()
+                        )
+                    )
+                    if not include_header:
+                        # Make sure the output starts with something that's not
+                        # a comment or indentation or blank line
+                        self.assertRegex(output.splitlines(keepends=True)[0], r"^[^#\s]+")
```

Comment:
I think this migration should always start with `from django.db import migrations` and I would find that assertion much simpler.

Reply:
It would be a simpler assertion, and not hard to fix if that behavior ever changed, but I think it does generically test that the migration starts with code instead of a `#` comment.

Reply:
Yeah it would inadvertently pass if the header were a `"""` docstring or something.